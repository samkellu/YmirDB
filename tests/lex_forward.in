set a 123
set b a
set c b
forward c
bye
