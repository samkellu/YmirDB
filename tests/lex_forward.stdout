==762== Memcheck, a memory error detector
==762== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==762== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==762== Command: ./ymirdb
==762== 
==762== error calling PR_SET_PTRACER, vgdb might block
> ok

> ok

> ok

> b, a2len0 a
1 (null)
==762== Invalid read of size 1
==762==    at 0x483FED4: strcmp (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==762==    by 0x10C5EA: lex_sort (ymirdb.c:681)
==762==    by 0x10C87D: command_forward (ymirdb.c:725)
==762==    by 0x10D32C: main (ymirdb.c:860)
==762==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==762== 
==762== 
==762== Process terminating with default action of signal 11 (SIGSEGV)
==762==  Access not within mapped region at address 0x0
==762==    at 0x483FED4: strcmp (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==762==    by 0x10C5EA: lex_sort (ymirdb.c:681)
==762==    by 0x10C87D: command_forward (ymirdb.c:725)
==762==    by 0x10D32C: main (ymirdb.c:860)
==762==  If you believe this happened as a result of a stack
==762==  overflow in your program's main thread (unlikely but
==762==  possible), you can try to increase the size of the
==762==  main thread stack using the --main-stacksize= flag.
==762==  The main thread stack size used in this run was 8388608.
==762== 
==762== HEAP SUMMARY:
==762==     in use at exit: 664 bytes in 11 blocks
==762==   total heap usage: 31 allocs, 20 frees, 2,104 bytes allocated
==762== 
==762== LEAK SUMMARY:
==762==    definitely lost: 0 bytes in 0 blocks
==762==    indirectly lost: 0 bytes in 0 blocks
==762==      possibly lost: 0 bytes in 0 blocks
==762==    still reachable: 664 bytes in 11 blocks
==762==         suppressed: 0 bytes in 0 blocks
==762== Reachable blocks (those to which a pointer was found) are not shown.
==762== To see them, rerun with: --leak-check=full --show-leak-kinds=all
==762== 
==762== For lists of detected and suppressed errors, rerun with: -s
==762== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
